NitroxPatcher/Patches/LargeWorldOctreeCompiler_RequestAbort_Patch.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Threading;
+using Harmony;
+using NitroxClient.Communication.Abstract;
+using NitroxClient.GameLogic;
+using NitroxModel.Core;
+using NitroxModel.Helper;
+
+namespace NitroxPatcher.Patches
+{
+    public class LargeWorldOctreeCompiler_RequestAbort_Patch : NitroxPatch
+    {
+        public static readonly Type TARGET_CLASS = typeof(LargeWorldOctreeCompiler);
+        public static readonly MethodInfo TARGET_METHOD = TARGET_CLASS.GetMethod("RequestAbort", BindingFlags.Public | BindingFlags.Instance);
+
+        public static void Prefix()
+        {
+            IMultiplayerSession multiplayerSession = NitroxServiceLocator.LocateService<IMultiplayerSession>();
+            multiplayerSession.Disconnect();
```

Comment:
The session already stops when the `XMenu` scene is loaded. Why is this patch needed?

Reply:
in some cases you quit game and in normal cleanup delete all item on the server

My Log
Nitrox] I: Disconnected from server.
[Nitrox] I: Client Disconnected. < LargeWorldOctreeCompiler_RequestAbort_Patch.cs  RUN
[Nitrox] I: Equipment Remove________________________________________________________
[Nitrox] I: Cannot send packet to a closed connection.
[Nitrox] I: Cannot send packet to a closed connection.
[Nitrox] I: Cannot send packet to a closed connection.
[Nitrox] I: Cannot send packet to a closed connection.
[Nitrox] I: Cannot send packet to a closed connection.
[Nitrox] I: Equipment Remove________________________________________________________
[Nitrox] I: Cannot send packet to a closed connection.
[Nitrox] I: Equipment Remove________________________________________________________
[Nitrox] I: Cannot send packet to a closed connection.
[Nitrox] I: Equipment Remove________________________________________________________
[Nitrox] I: Cannot send packet to a closed connection.
[Nitrox] I: Cannot send packet to a closed connection.
[Nitrox] I: Cannot send packet to a closed connection.
[Nitrox] I: Cannot send packet to a closed connection.
[Nitrox] I: Cannot send packet to a closed connection.

Reply:
I think the issues is that connection gets closed too soon, not too late.

When I removed the `Player_OnDestroy_Patch` the connection stays open long enough for the packets (send by other code patches) to get through.

I noticed this is from a commit that fixes a [player inventory wipe](https://github.com/SubnauticaNitrox/Nitrox/commit/ab58b2f9e80c6a8e4ddf09321cb8c25d7c8991b2) issue but I think that the fix actually breaks the connection which prevents player inventory wipe.

Reply:
i want packet no send

Reply:
The packet shouldn't be sent if you don't want it.

Reply:
it lot packet to add check mire easy call disconnect

Reply:
Have you tried reordering this code (Multiplayer.StopCurrentSession):

        public void StopCurrentSession()
        {
            SceneManager.sceneLoaded -= SceneManager_sceneLoaded;
			
            if (multiplayerSession.CurrentState.CurrentStage != MultiplayerSessionConnectionStage.Disconnected)
            {
                multiplayerSession.Disconnect();
            }
			
            OnAfterMultiplayerEnd?.Invoke();

            //Always do this last.
            NitroxServiceLocator.EndCurrentLifetimeScope();
        }

The `OnAfterMultiplayerEnd?.Invoke();` restores the code patches that are sending packets. Maybe it's fixed if you restore the patches *before* disconnecting.

Anyway, just forcing to disconnect when packets are still being requested/send isn't a proper solution. I think you should either remove the code patch in this PR [and revert this patch](https://github.com/SubnauticaNitrox/Nitrox/commit/ab58b2f9e80c6a8e4ddf09321cb8c25d7c8991b2) or find a better solution.